packages/widget/src/store/callbacks.ts

```diff
@@ -0,0 +1,41 @@
+import { create } from 'zustand';
+
+type OnWalletConnected = {
+  walletName: string;
+  chainId: string;
+  address?: string;
+};
+
+type OnWalletDisconnected = {
+  chainType?: string;
+};
+
+type OnTransactionBroadcasted = {
+  txHash: string;
+  chainId: string;
+  explorerLink: string;
+};
+
+type OnTransactionComplete = {
+  txHash: string;
+  chainId: string;
+  explorerLink: string;
+};
+
+type OnTransactionFailed = {
+  error: string;
+};
+
+export type WalletCallbackStore = {
+  onWalletConnected?: (props: OnWalletConnected) => void;
+  onWalletDisconnected?: (props: OnWalletDisconnected) => void;
+  onTransactionBroadcasted?: (props: OnTransactionBroadcasted) => void;
+  onTransactionComplete?: (props: OnTransactionComplete) => void;
+  onTransactionFailed?: (props: OnTransactionFailed) => void;
+};
+
+export const defaultValues: WalletCallbackStore = {};
+
+export const useCallbackStore = create<WalletCallbackStore>()(
+  () => defaultValues
+);
```

Comment:
callbacks can go anywhere in the codebase and I don't want to prop drill them everywhere, I think it makes sense to use a store for this

Reply:
yeah looks good